### ISSUE TITLE
mintupdate forcefully removes cinnamon with a third party repository

### ISSUE THREAD

**Issue by tanev** [linuxmint/mintupdate#636]
Hey guys,

First of all, thanks a lot for the great distribution and all the efforts around it. It is amazing how far it got compared to where I've started using it more than 10 years ago! :)

I've upgraded to Mint 20 from 19.03 a month ago, but due to the pandemic I've been using the machine almost constantly while working from home so I hadn't had the need to put it to sleep.

The bug I am reporting happened twice right after the node is awaken from sleep.

What's happening is that as it seems mintupdate executes the following (snippet from the attached apt's history.log):

```
Commandline: apt-get install --assume-yes --option Dpkg::Options::=--force-confnew libgles2-mesa mesa-opencl-icd libxatracker2 libgbm1 libgl1-mesa-glx mesa-vdpau-drivers mesa-vulkan-drivers
Start-Date: 2020-08-08  08:03:43
Commandline: apt-get install --assume-yes --option Dpkg::Options::=--force-confnew libgles2-mesa mesa-opencl-icd libxatracker2 libgbm1 libgl1-mesa-glx mesa-vdpau-drivers mesa-vulkan-drivers
Upgrade: libgles2-mesa:amd64 (20.3~git2008070730.c5b231~oibaf~f, 20.3~git2008071930.0cdd41~oibaf~f), mesa-opencl-icd:amd64 (20.3~git2008070730.c5b231~oibaf~f, 20.3~git2008071930.0cdd41~oibaf~f), libxatracker2:amd64 (20.3~git2008070730.c5b231~oibaf~f, 20.3~git2008071930.0cdd41~oibaf~f), libgbm1:amd64 (20.3~git2008070730.c5b231~oibaf~f, 20.3~git2008071930.0cdd41~oibaf~f), libgl1-mesa-glx:amd64 (20.3~git2008070730.c5b231~oibaf~f, 20.3~git2008071930.0cdd41~oibaf~f), mesa-vdpau-drivers:amd64 (20.3~git2008070730.c5b231~oibaf~f, 20.3~git2008071930.0cdd41~oibaf~f), mesa-vulkan-drivers:amd64 (20.3~git2008070730.c5b231~oibaf~f, 20.3~git2008071930.0cdd41~oibaf~f)
Remove: libgoa-backend-1.0-1:amd64 (3.36.0-linuxmint1), libchamplain-gtk-0.12-0:amd64 (0.12.20-1), xserver-xorg-input-all:amd64 (1:7.7+19ubuntu14), libmpv1:amd64 (0.32.0-1ubuntu1), geary:amd64 (3.36.1-1), xserver-xorg:amd64 (1:7.7+19ubuntu14), gnome-calendar:amd64 (3.36.1-linuxmint1), qml-module-qtquick-privatewidgets:amd64 (5.9.5-0ubuntu1), libcogl20:amd64 (1.22.6-1), qt5-image-formats-plugins:amd64 (5.9.5-0ubuntu1), libqt5svg5:amd64 (5.9.5-0ubuntu1), libclutter-1.0-0:amd64 (1.26.4+dfsg-1), libedataserverui-1.2-2:amd64 (3.36.3-0ubuntu1.1), qml-module-qtquick-window2:amd64 (5.9.5-0ubuntu1.1), libcegui-mk2-0.8.7:amd64 (0.8.7-6ubuntu2), libclutter-gst-3.0-0:amd64 (3.0.27-1), libcegui-mk2-dev:amd64 (0.8.7-6ubuntu2), steam-launcher:amd64 (1:1.0.0.64), libepoxy-dev:amd64 (1.5.4-1), libegl1:amd64 (1.3.1-1), xserver-xorg-core:amd64 (2:1.20.8-2ubuntu2.2), gir1.2-clutter-1.0:amd64 (1.26.4+dfsg-1), qml-module-qtquick2:amd64 (5.9.5-0ubuntu1.1), zoom:amd64 (3.5.361976.0301), libegl1-mesa-dev:amd64 (20.3~git2008070730.c5b231~oibaf~f), libcogl-path20:amd64 (1.22.6-1), libegl-mesa0:amd64 (20.3~git2008070730.c5b231~oibaf~f), libsdl2-image-dev:amd64 (2.0.5+dfsg1-2), libgles-dev:amd64 (1.3.1-1), yelp:amd64 (3.36.0-linuxmint1), libyelp0:amd64 (3.36.0-linuxmint1), qml-module-qtquick-controls:amd64 (5.9.5-0ubuntu1), gir1.2-meta-muffin-0.0:amd64 (4.6.2+ulyana), mintsystem:amd64 (8.4.7), xserver-xorg-input-libinput:amd64 (0.29.0-1), qml-module-qtquick-layouts:amd64 (5.9.5-0ubuntu1.1), libqt5widgets5:amd64 (5.9.5+dfsg-0ubuntu2.5), libcogl-pango20:amd64 (1.22.6-1), metacity:amd64 (1:3.36.1-1), libegl1-mesa:amd64 (20.3~git2008070730.c5b231~oibaf~f), evolution-data-server:amd64 (3.36.3-0ubuntu1.1), qml-module-qtquick-dialogs:amd64 (5.9.5-0ubuntu1), libsdl2-dev:amd64 (2.0.10+dfsg1-3), firetools:amd64 (0.9.50-1), xserver-xorg-input-wacom:amd64 (1:0.39.0-0ubuntu1), mint-meta-cinnamon:amd64 (2020.07.09), smplayer:amd64 (18.2.2~ds0-1), gstreamer1.0-clutter:amd64 (2.0.18-1), boinc-manager:amd64 (7.16.6+dfsg-1), libwebkitgtk-3.0-0:amd64 (2.4.11-3ubuntu3), libqt5printsupport5:amd64 (5.9.5+dfsg-0ubuntu2.5), libmuffin0:amd64 (4.6.2+ulyana), libwebkitgtk-1.0-0:amd64 (2.4.11-3ubuntu3), cinnamon-control-center:amd64 (4.6.1+ulyana), ubuntu-system-adjustments:amd64 (2020.07.29), mint-meta-codecs:amd64 (2020.07.09), libegl-dev:amd64 (1.3.1-1), gir1.2-cogl-1.0:amd64 (1.22.6-1), pix:amd64 (2.4.11+ulyana), libqt5gui5:amd64 (5.9.5+dfsg-0ubuntu2.5), celluloid:amd64 (0.19-linuxmint1+ulyana), libchamplain-0.12-0:amd64 (0.12.20-1), libqt5x11extras5:amd64 (5.9.5-0ubuntu1), mint-meta-core:amd64 (2020.07.09), libglvnd-dev:amd64 (1.3.1-1), libgstreamer-gl1.0-0:amd64 (1.16.2-4), gir1.2-webkit-3.0:amd64 (2.4.11-3ubuntu3), libwayland-egl1-mesa:amd64 (20.3~git2008070730.c5b231~oibaf~f), libdevhelp-3-6:amd64 (3.36.1-1), libclutter-gst-2.0-0:amd64 (2.0.18-1), libqt5webkit5:amd64 (5.212.0~alpha2-7ubuntu1), libgles2-mesa-dev:amd64 (20.3~git2008070730.c5b231~oibaf~f), mintupgrade:amd64 (2020.07.11), zenity:amd64 (3.32.0-5), libogre-1.9.0v5:amd64 (1.9.0+dfsg1-12build1), libwebkit2gtk-4.0-37:amd64 (2.28.4-0ubuntu0.20.04.1), gnome-online-accounts:amd64 (3.36.0-linuxmint1), xed:amd64 (2.6.2+ulyana), gir1.2-gst-plugins-base-1.0:amd64 (1.16.2-4), libtotem0:amd64 (3.34.1-2ubuntu2), libgl1-mesa-dev:amd64 (20.3~git2008070730.c5b231~oibaf~f), gir1.2-totem-1.0:amd64 (3.34.1-2ubuntu2), xorg:amd64 (1:7.7+19ubuntu14), libwxgtk-webview3.0-gtk3-0v5:amd64 (3.0.4+dfsg-15build1), gnome-session-bin:amd64 (3.36.0-2ubuntu1), gir1.2-coglpango-1.0:amd64 (1.22.6-1), caribou:amd64 (0.4.21-7), cinnamon:amd64 (4.6.6+ulyana), gstreamer1.0-vaapi:amd64 (1.16.2-2), libqt5quick5:amd64 (5.9.5-0ubuntu1.1), cinnamon-common:amd64 (4.6.6+ulyana), gir1.2-gtkclutter-1.0:amd64 (1.8.4-4), qml-module-qtgraphicaleffects:amd64 (5.9.5-0ubuntu1), libclutter-gtk-1.0-0:amd64 (1.8.4-4), devhelp:amd64 (3.36.1-1), xviewer-plugins:amd64 (1.2.0+ulyana)
End-Date: 2020-08-08  08:04:19
```

I found the apt's output in the attached mintupdate.log, thus the bug report here.

Unsure why does it happen only on boot, but my guess would be that the tunnel I'm using (an sshuttle), gets stuck when its TCP session expires through the night, leading to loss of connectivity - this might be related.

I have, of course a snapshot to restore to, but would be nice if I don't have to :)

Some details about the hardware and the installation:

```
[GIT master mintupdate ]$ inxi -CMG
Machine:
  Type: Desktop Mobo: ASUSTeK model: PRIME X470-PRO v: Rev X.0x 
  serial: <superuser/root required> UEFI [Legacy]: American Megatrends 
  v: 5406 date: 11/13/2019 
CPU:
  Topology: 8-Core model: AMD Ryzen 7 2700X bits: 64 type: MT MCP 
  L2 cache: 4096 KiB 
  Speed: 1985 MHz min/max: 2200/3700 MHz Core speeds (MHz): 1: 1975 2: 1886 
  3: 2050 4: 2018 5: 2184 6: 2180 7: 2192 8: 2195 9: 1970 10: 1890 11: 1986 
  12: 2199 13: 2189 14: 2177 15: 2187 16: 1992 
Graphics:
  Device-1: AMD Navi 10 [Radeon RX 5600 OEM/5600 XT / 5700/5700 XT] 
  driver: amdgpu v: kernel 
  Display: x11 server: X.Org 1.20.8 driver: modesetting 
  resolution: 2560x1440~60Hz 
  OpenGL: renderer: AMD Radeon RX 5700 XT (NAVI10 DRM 3.37.0 
  5.7.8-050708-lowlatency LLVM 10.0.1) 
  v: 4.6 Mesa 20.3.0-devel (git-c5b2319 2020-08-07 focal-oibaf-ppa) 
```

OS version:
```
[GIT master mintupdate ]$ cat /etc/linuxmint/info 
RELEASE=20
CODENAME=ulyana
EDITION="Cinnamon"
DESCRIPTION="Linux Mint 20 Ulyana"
DESKTOP=Gnome
TOOLKIT=GTK
NEW_FEATURES_URL=https://www.linuxmint.com/rel_ulyana_cinnamon_whatsnew.php
RELEASE_NOTES_URL=https://www.linuxmint.com/rel_ulyana_cinnamon.php
USER_GUIDE_URL=https://www.linuxmint.com/documentation.php
GRUB_TITLE=Linux Mint 20 Cinnamon
```

mintupdate version:

```
[GIT master mintupdate ]$ dpkg -l mintupdate 
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name           Version      Architecture Description
+++-==============-============-============-=================================
ii  mintupdate     5.6.9        all          The Linux Mint Update Manager
```

The custom repository that might be leading to the issue is [oibaf](https://launchpad.net/~oibaf/+archive/ubuntu/graphics-drivers), which unfortunately is required due to the GPU I'm using.

EDIT:~Haven't tried what would happen if connectivity is not lost when the system is awaken, which I will do in a moment when I restore back to the last timeshift snapshot and will report back here.~ Doesn't seem to be related to the loss of connectivity, easily reproduced with just a simple suspend/resume.

[mintupdate.log](https://github.com/linuxmint/mintupdate/files/5044764/mintupdate.log)
[history.log](https://github.com/linuxmint/mintupdate/files/5044766/history.log)

EDIT2: Forgot to mention that the automatic updates are disabled in update manager, unsure if this is related:
![update_manager_settings](https://user-images.githubusercontent.com/4588129/89704413-4a00b380-d95c-11ea-8737-3bcdd181395f.png)

EDIT3: Happened even with `chmod -x /usr/bin/mintupdate`... this must be some cron/systemd recurring job.

EDIT4: Yep, if this happen to you comment out this line `/usr/bin/mintupdate-cli upgrade --refresh-cache --yes > /var/log/mintupdate.log 2>&1` in `/etc/cron.daily/mintupdate`.

Let me know if any other info will be required and thanks in advance for looking into this.

Best,
T.

**Comment by tanev** [comment]
So, not really sure if this is a bug or rather an implication from the custom oibaf and the default mintupdate daily cron action, update manager shows only this mesa update:
![image](https://user-images.githubusercontent.com/4588129/89706732-02d0ed80-d971-11ea-9a3f-90af0a1cf385.png)

an attempt to perform `apt-get update && apt-get upgrade` tries to:

```
$ sudo apt-get upgrade 
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Calculating upgrade... Done
The following packages have been kept back:
  clementine firetools gpsbabel keepassx libaccounts-qt5-1 libegl-mesa0 libgbm1 libgl1-mesa-dri:i386 libgl1-mesa-dri libglapi-mesa:i386 libglapi-mesa libglx-mesa0
  libglx-mesa0:i386 libkf5archive5 libqt5concurrent5 libqt5core5a libqt5dbus5 libqt5gui5 libqt5network5 libqt5positioning5 libqt5printsupport5 libqt5qml5 libqt5quick5
  libqt5script5 libqt5sensors5 libqt5sql5 libqt5sql5-sqlite libqt5svg5 libqt5test5 libqt5webchannel5 libqt5webkit5 libqt5widgets5 libqt5x11extras5 libqt5xml5 libsignon-extension1
  libsignon-plugins-common1 libsignon-qt5-1 qml-module-qtgraphicaleffects qml-module-qtquick-controls qml-module-qtquick-dialogs qml-module-qtquick-layouts
  qml-module-qtquick-privatewidgets qml-module-qtquick-window2 qml-module-qtquick2 qt5-image-formats-plugins qtbase5-dev-tools signon-plugin-oauth2 signon-plugin-password signond
  smplayer smtube
The following packages will be upgraded:
  libegl1-mesa libegl1-mesa-dev libgl1-mesa-dev libgl1-mesa-glx libgles2-mesa libgles2-mesa-dev libwayland-egl1-mesa libxatracker2 mesa-opencl-icd mesa-vdpau-drivers
  mesa-vulkan-drivers
11 upgraded, 0 newly installed, 0 to remove and 51 not upgraded.
Need to get 19,8 MB of archives.
After this operation, 4096 B disk space will be freed.
Do you want to continue? [Y/n] n
Abort.
```

but an `apt-get dist-upgrade` would lead to the same outcome as initially reported:

```
$ sudo apt-get dist-upgrade 
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Calculating upgrade... Done
The following packages were automatically installed and are no longer required:
  acpi crudini dcraw devhelp-common fonts-baekmuk fonts-cegui fonts-junicode fonts-jura gir1.2-keybinder-3.0 gir1.2-nma-1.0 gir1.2-totemplparser-1.0 liballegro4.4 libamtk-5-0
  libamtk-5-common libcegui-mk2-data libdazzle-1.0-0 libdevil-dev libdevil1c2 libedata-cal-2.0-1 libfolks25 libfreeimage-dev libfreeimage3 libgles1 libglew2.1 libgmime-3.0-0
  libgrilo-0.3-0 libhal1-flash libjxr0 libmng-dev libopengl-dev libopengl0 libqt5script5 libqt5webchannel5 libsdl2-image-2.0-0 libsilly libsndio-dev libtinyxml2.6.2v5
  libva-wayland2 libwayland-dev libwoff1 libxcb-res0 libxcb-xtest0 libxss-dev libxt-dev libxxf86vm-dev libytnef0 lua-expat lua-expat-dev mintreport pix-data
  python-tinycss2-common python3-iniparse python3-macaroonbakery python3-pam python3-pymacaroons python3-pyxattr python3-rfc3339 python3-tinycss2 python3-tz python3-webencodings
  rtmpdump streamer ubuntu-dbgsym-keyring x11proto-scrnsaver-dev x11proto-xf86vidmode-dev xawtv-plugins xed-common youtube-dl
Use 'sudo apt autoremove' to remove them.
The following packages will be REMOVED:
  boinc-manager caribou celluloid cinnamon cinnamon-common cinnamon-control-center devhelp evolution-data-server firetools geary gir1.2-clutter-1.0 gir1.2-cogl-1.0
  gir1.2-coglpango-1.0 gir1.2-gst-plugins-base-1.0 gir1.2-gtkclutter-1.0 gir1.2-meta-muffin-0.0 gir1.2-totem-1.0 gir1.2-webkit-3.0 gnome-calendar gnome-online-accounts
  gnome-session-bin gstreamer1.0-clutter gstreamer1.0-vaapi libcegui-mk2-0.8.7 libcegui-mk2-dev libchamplain-0.12-0 libchamplain-gtk-0.12-0 libclutter-1.0-0 libclutter-gst-2.0-0
  libclutter-gst-3.0-0 libclutter-gtk-1.0-0 libcogl-pango20 libcogl-path20 libcogl20 libdevhelp-3-6 libedataserverui-1.2-2 libegl-dev libegl-mesa0 libegl1 libegl1-mesa
  libegl1-mesa-dev libepoxy-dev libgl1-mesa-dev libgles-dev libgles2-mesa-dev libglvnd-dev libgoa-backend-1.0-1 libgstreamer-gl1.0-0 libmpv1 libmuffin0 libogre-1.9.0v5 libqt5gui5
  libqt5printsupport5 libqt5quick5 libqt5svg5 libqt5webkit5 libqt5widgets5 libqt5x11extras5 libsdl2-dev libsdl2-image-dev libtotem0 libwayland-egl1-mesa libwebkit2gtk-4.0-37
  libwebkitgtk-1.0-0 libwebkitgtk-3.0-0 libwxgtk-webview3.0-gtk3-0v5 libyelp0 metacity mint-meta-cinnamon mint-meta-codecs mint-meta-core mintsystem mintupgrade pix
  qml-module-qtgraphicaleffects qml-module-qtquick-controls qml-module-qtquick-dialogs qml-module-qtquick-layouts qml-module-qtquick-privatewidgets qml-module-qtquick-window2
  qml-module-qtquick2 qt5-image-formats-plugins smplayer steam-launcher ubuntu-system-adjustments xed xorg xserver-xorg xserver-xorg-core xserver-xorg-input-all
  xserver-xorg-input-libinput xserver-xorg-input-wacom xviewer-plugins yelp zenity zoom
The following packages have been kept back:
  clementine gpsbabel keepassx libaccounts-qt5-1 libgl1-mesa-dri:i386 libgl1-mesa-dri libglapi-mesa:i386 libglapi-mesa libglx-mesa0 libglx-mesa0:i386 libkf5archive5
  libqt5concurrent5 libqt5core5a libqt5dbus5 libqt5network5 libqt5positioning5 libqt5qml5 libqt5script5 libqt5sensors5 libqt5sql5 libqt5sql5-sqlite libqt5test5 libqt5webchannel5
  libqt5xml5 libsignon-extension1 libsignon-plugins-common1 libsignon-qt5-1 qtbase5-dev-tools signon-plugin-oauth2 signon-plugin-password signond smtube
The following packages will be upgraded:
  libgbm1 libgl1-mesa-glx libgles2-mesa libxatracker2 mesa-opencl-icd mesa-vdpau-drivers mesa-vulkan-drivers
7 upgraded, 0 newly installed, 96 to remove and 32 not upgraded.
Need to get 19,5 MB of archives.
After this operation, 585 MB disk space will be freed.
Do you want to continue? [Y/n] n
Abort.
```

**Comment by tanev** [comment]
Unlikely to be related to the mintupdate at all, seems to be a post-upgrade (from 19) leftover dependency issue, that revealed itself just now when a package was updated. Cannot be reproduced on a clean installation with Mint 20 and oibaf. Removing all packages marked to be held and removing/installing back oibaf returned back to normal state.
Apologies I've jumped the gun, but it was still a big surprise that the whole desktop environment gotten removed :)
